msm/src/proof.rs

```diff
@@ -66,21 +72,21 @@ impl<const N: usize, F: Clone> ColumnEvaluations<F> for ProofEvaluations<N, F> {
 }
 
 #[derive(Debug, Clone)]
-pub struct ProofCommitments<const N: usize, G: KimchiCurve> {
+pub struct ProofCommitments<const N: usize, G: KimchiCurve, ID: LookupTableID> {
     /// Commitments to the N columns of the circuits, also called the 'witnesses'.
     /// If some columns are considered as public inputs, it is counted in the witness.
     pub(crate) witness_comms: Witness<N, PolyComm<G>>,
     /// Commitments to the polynomials used by the lookup argument.
     /// The values contains the chunked polynomials.
-    pub(crate) mvlookup_comms: Option<LookupProof<PolyComm<G>>>,
+    pub(crate) mvlookup_comms: Option<LookupProof<PolyComm<G>, ID>>,
     /// Commitments to the quotient polynomial.
     /// The value contains the chunked polynomials.
     pub(crate) t_comm: PolyComm<G>,
 }
 
 #[derive(Debug, Clone)]
-pub struct Proof<const N: usize, G: KimchiCurve, OpeningProof: OpenProof<G>> {
-    pub(crate) proof_comms: ProofCommitments<N, G>,
-    pub(crate) proof_evals: ProofEvaluations<N, G::ScalarField>,
+pub struct Proof<const N: usize, G: KimchiCurve, OpeningProof: OpenProof<G>, ID: LookupTableID> {
```

Comment:
This all starts to become quite heavy. I'm thinking we could do a trait `ProofParams` with a bunch of associated types instead, and just pass that around. Similar to how arkworks does for a `Circuit` if I remember correctly?